src/main/java/org/broadinstitute/hellbender/tools/walkers/realignmentfilter/FilterAlignmentArtifacts.java

```diff
@@ -0,0 +1,274 @@
+ *     step-by-step description of the workflow and <a href="https://software.broadinstitute.org/gatk/documentation/article?id=11127">Article#11127</a>
+ *     for an overview of what traditional somatic calling entails. For the latest pipeline scripts, see the
+ *     <a href="https://github.com/broadinstitute/gatk/tree/master/scripts/mutect2_wdl">Mutect2 WDL scripts directory</a>.
+ * </p>
+ * <p>
+ *     The bam input to this tool should be the reassembly bamout produced by HaplotypeCaller or Mutect2 in the process of generating
+ *     the input callset.  The original bam will also work but might fail to filter some indels.  The reference passed with the -R argument
+ *     but be the reference to which the input bam was realigned.  This does not need to correspond to the reference of the BWAS-MEM
+ *     index image.  The latter should be derived from the best available reference, for example hg38 in humans as of February 2018.
+ * </p>
+ *
+ * <h3>Usage example</h3>
+ * <pre>
+ * gatk FilterAlignmentArtifacts \
+ *   -R hg19.fasta
+ *   -V somatic.vcf.gz \
+ *   -I somatic_bamout.bam \
+ *   --bwa-mem-index-image hg38.index_image \
+ *   -O filtered.vcf.gz
+ * </pre>
+ *
+ */
+@CommandLineProgramProperties(
+        summary = "Filter alignment artifacts from a vcf callset.",
+        oneLineSummary = "Filter alignment artifacts from a vcf callset.",
+        programGroup = VariantFilteringProgramGroup.class
+)
+@DocumentedFeature
+@ExperimentalFeature
+public class FilterAlignmentArtifacts extends VariantWalker {
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName=StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc="The output filtered VCF file", optional=false)
+    private final String outputVcf = null;
+
+    public static final int DEFAULT_INDEL_START_TOLERANCE = 5;
+    public static final String INDEL_START_TOLERANCE_LONG_NAME = "indel-start-tolerance";
+    @Argument(fullName = INDEL_START_TOLERANCE_LONG_NAME,
+            doc="Max distance between indel start of aligned read in the bam and the variant in the vcf", optional=true)
+    private int indelStartTolerance = DEFAULT_INDEL_START_TOLERANCE;
+
+    public static final int DEFAULT_FRAGMENT_SIZE = 1000;
+    public static final String FRAGMENT_SIZE_LONG_NAME = "fragment-size";
+    @Argument(fullName = FRAGMENT_SIZE_LONG_NAME,
+            doc="Distance away from variant to look for reads' mates.", optional=true)
+    private int fragmentSize = DEFAULT_FRAGMENT_SIZE;
+
+    public static final int DEFAULT_MIN_MAPPING_QUALITY = 20;
+    public static final String MIN_MAPPING_QUALITY_LONG_NAME = "min-mapping-quality";
+    @Argument(fullName = MIN_MAPPING_QUALITY_LONG_NAME,
+            doc="Ignore reads with mapping quality less than this", optional=true)
+    private int minMappingQuality = DEFAULT_MIN_MAPPING_QUALITY;
```

Comment:
This can be a default `ReadFilter` ([MappingQualityReadFilter](https://github.com/broadinstitute/gatk/blob/master/src/main/java/org/broadinstitute/hellbender/engine/filters/MappingQualityReadFilter.java)), which can be tweaked by the user as a plugin (delete the filter, change the minimum MQ, set a maximum one, etc). Otherwise, the filter can be applied twice!